src/main/java/com/sonyericsson/hudson/plugins/gerrit/trigger/dependency/GerritDependencyAction.java

```diff
@@ -0,0 +1,74 @@
+package com.sonyericsson.hudson.plugins.gerrit.trigger.dependency;
+
+import hudson.EnvVars;
+import hudson.model.AbstractBuild;
+import hudson.model.EnvironmentContributingAction;
+import hudson.model.InvisibleAction;
+import hudson.model.Run;
+
+import javax.annotation.Nonnull;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Adds Action that stores the data about dependency jobs.
+ */
+public class GerritDependencyAction extends InvisibleAction implements EnvironmentContributingAction {
+    @Nonnull
+    private final List<String> deps;
+
+    /**
+     * Saves the important information about the parent runs to be persistent.
+     * Stores string to keep the data even if original execution of parent job would be removed from history.
+     * @param runs List of runs it depend on
+     */
+    public GerritDependencyAction(List<Run> runs) {
+        deps = new ArrayList<String>(runs.size());
+        for (Run run : runs) {
+            deps.add(run.getParent().getFullName() + "#" + run.getNumber() + "#" + run.getResult());
```

Comment:
See if you can use [getExternalizableId](http://javadoc.jenkins-ci.org/hudson/model/Run.html#getExternalizableId--)